elastic/kibana
# [Logs Data Telemetry] Report telemetry events to telemetry service

## Summary

The PR defines the schema and  registers the [Usage Collector](https://docs.elastic.dev/telemetry/collection/snapshot-telemetry#creating-and-registering-usage-collector) to report Logs Data telemetry events introduced in https://github.com/elastic/kibana/pull/189380. 

This will introduce a telemetry [subsection](https://docs.elastic.dev/telemetry/kibana-snapshot-subsections) under `stats.stack_stats.kibana.plugins.logs_data`. _This may likely be promoted to `stats.logs_data` given the nature of the data and as the schema matures._

### How to test

1. Make sure the task (task id: `logs-data-telemetry:collect-and-report-task`) has been run at least once. [`TELEMETRY_TASK_INTERVAL`](https://github.com/elastic/kibana/pull/192868/files#diff-61ba101eb2c95a27ebb1e45700879c668b9768572b1a3b564780b5781c557856R15) can be changed e.g. to 1 minutes to run it immediately.
2. The queued data can be viewed by invoking the _stats api in browser console and inspecting the key `[0].stats.stack_stats.kibana.plugins.logs_data`:

```javascript
await (await fetch("/<xyz>/internal/telemetry/clusters/_stats", {"body": JSON.stringify({"unencrypted":true,"refreshCache":true}), "method": "POST", "headers": {"kbn-xsrf": "x", "elastic-api-version": "2", "x-elastic-internal-origin": "x"}})).json()
```

  Once the task (task id: `logs-data-telemetry:collect-and-report-task`) is run, the captured telemetry data can also be viewed in UI under Kibana -> Advanced Settings -> Global Settings -> Usage collection -> _click_ "cluster data".
  ![image](https://github.com/user-attachments/assets/a232c329-7d3b-451b-8c84-149264423002)
